xknx/telegram/apci.py

```diff
@@ -0,0 +1,207 @@
+"""
+Module for serialization and deserialization of APCI payloads.
+
+APCI stands for Application Layer Protocol Control Information.
+
+An APCI payload contains a service and payload. For example, a GroupValueWrite
+is a service that takes a DPT as a value.
+"""
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Optional, Type, Union
+
+from xknx.dpt import DPTArray, DPTBinary
+from xknx.exceptions import ConversionError
+
+
+def encode_cmd_and_payload(
+    cmd: "APCICommand",
+    encoded_payload: int = 0,
+    appended_payload: Optional[bytes] = None,
+) -> bytes:
+    """Encode cmd and payload."""
+    if appended_payload is None:
+        appended_payload = bytes()
+
+    data = bytearray(
+        [
+            (cmd.value >> 8) & 0xFF,
+            (cmd.value & 0xFF) | (encoded_payload & DPTBinary.APCI_BITMASK),
+        ]
+    )
+    data.extend(appended_payload)
+    return data
+
+
+class APCICommand(Enum):
+    """Enum class for APCI services."""
+
+    GROUP_READ = 0x0000
+    GROUP_RESPONSE = 0x0040
+    GROUP_WRITE = 0x0080
+
+    ESCAPE = 0x03C0
+
+
+class APCIExtendedCommand(Enum):
+    """Enum class for extended APCI services."""
+
+
+class APCI(ABC):
+    """
+    Base class for ACPI services.
+
+    This base class is only the interface for the derived classes.
+    """
+
+    code: APCICommand = APCICommand.ESCAPE
```

Comment:
```suggestion
    @property
    @classmethod
    @abstractmethod
    def code(cls) -> APCICommand
        return NotImplementedError
```
This is a wild guess, but would be nice if it works.

Reply:
I guess that changing it to `code: APCICommand` is sufficient as well.

Reply:
oh... I didn't know this worked 🤣 beautiful

Reply:
I think I tried this before, but because it wasn't an ABC, it did not work (with mypy).

Reply:
Ah, damn. It did not work :-P

Reply:
When I fix it for Mypy, then pylint complains. Guess it needs to be silenced somehow.

Reply:
`code: ClassVar[APCICommand] = cast(APCICommand, None)` did the trick.